data/35e/wizards_of_the_coast/supplement/dragon_magic/dm_abilities.lst

```diff
@@ -130,11 +130,15 @@ Devour Spell	KEY:Spelleater ~ Devour Spell			CATEGORY:Special Ability	TYPE:Spell
+Dragontouched			KEY:Dragonfire Adept ~ Dragontouched			CATEGORY:Special Ability	TYPE:Dragonfire Adept Class Feature.SpecialQuality	SOURCEPAGE:p.26	DESC:At 1st level, you gain Dragontouched (see page 18) as a bonus feat.		ABILITY:FEAT|AUTOMATIC|Dragontouched
+#Applying suggested fix for Breath Effect
+Breath Effect			KEY:Dragonfire Adept ~ Breath Effect			CATEGORY:Special Ability	TYPE:Dragonfire Adept Class Feature.SpecialQuality	SOURCEPAGE:p.26	DESC:At 2nd, 5th, 10th, 12th, 15th, and 20th level, you can select one of the breath effects in Table 2-2: Dragonfire Adept Breath Effects for which you meet the minimum level prerequisite. These breath effects can alter your breath weapon's damage type or area, or apply a condition to targets in place of damage. Each time you use your breath weapon, you can choose to apply any one breath effect that you know. The chosen effect either replaces the normal fire damage dealt by your breath weapon or replaces the standard area of your breath weapon. Some effects can be applied only to a cone-shaped breath weapon, and others only to a line-shaped breath weapon. You can't apply more than one breath effect to your breath weapon unless the effect specifically states otherwise. Also, you can't apply the same breath effect to your breath weapon in two consecutive rounds (though you still can use your normal fire breath weapon every round). See page 77 in Chapter 3: Draconic Magic for full descriptions of the breath effects.		BONUS:ABILITYPOOL|Dragonfire Adept Breath Effect|1+(DragonfireAdeptLVL/5)+(DragonfireAdeptLVL>=12)
+#Scales AC bonus does not follow a mathematical progression.  - applying with PREVAR instead
+Scales				KEY:Dragonfire Adept ~ Scales					CATEGORY:Special Ability	TYPE:Extraordinary.Dragonfire Adept Class Feature.SpecialQuality	SOURCEPAGE:p.26	DESC:At 2nd level, your skin becomes thick and scaly, granting you a +2 bonus to your natural armor. The scales can be of any color or metallic hue; they are often (but not always) of a draconic hue that matches your outlook and alignment. This bonus improves to +3 at 8th level, to +4 at 13th level, and to +5 at 18th level. If you already have a natural armor bonus, use the higher of the two values.		DEFINE:ScalesArmorBonus|0	BONUS:COMBAT|AC|ScalesArmorBonus|TYPE=NaturalArmor		BONUS:VAR|ScalesArmorBonus|2		BONUS:VAR|ScalesArmorBonus|1|PREVARGTEQ:DragonfireAdeptLVL,8	BONUS:VAR|ScalesArmorBonus|1|PREVARGTEQ:DragonfireAdeptLVL,13	BONUS:VAR|ScalesArmorBonus|1|PREVARGTEQ:DragonfireAdeptLVL,18	
```

Comment:
Mate, 

If a limit or no stack is desired that is easy to handle.
DR:Magic/DR_DragonfireAdept
DEFINE:DR_DragonfireAdept|0
BONUS:VAR|Magic|2+if(DragonfireAdeptLVL>=16,3,0)

if the AC is a concern, do this:
BONUS:VAR|ScalesArmorBonus|2+min(((DragonfireAdeptLVL-3)/5),3)

I think that handles concerns of stacking and epic levels.

Best,